fail = (errorMessage) -> throw new Error(errorMessage)
